### ISSUE TITLE
Vespa for HA

### ISSUE THREAD

**Issue by JayGhiya** [unoplat/vespa-helm-charts#15]
After understanding the basic overview of vespa and https://github.com/vespa-engine/sample-apps/tree/master/examples/operations/multinode-HA/gke there are basic three categories of components that need to work together to make the ha setup happen:

1. Config Server
2. Admin node, feed container cluster, query container cluster and content node pods
3. Application package

Are there any explicit dependencies that any of the category has on each other? Are they truly independent so that they can be deployed in parallel ?

**Comment by kkraune** [comment]
Work note, from my Slack response:

To bootstrap your Helm efforts, take a look at https://github.com/vespa-engine/sample-apps/blob/master/examples/operations/multinode-HA/docker-compose.yaml - this has some very basic healthchecks, plus the required dependency from nodes to one of the config servers (node0 condition). it is not strictly required in start order but models the dependency of config quite well. Also see the startup sequence note in https://github.com/vespa-engine/sample-apps/tree/master/examples/operations/multinode-HA#start-the-admin-server, where the connectivity check (configurable) kicks in. https://docs.vespa.ai/en/operations-selfhosted/configuration-server.html is also an important read

In short, VESPA_CONFIGSERVERS must be set on all nodes. start the config server on the config server nodes, deploy the application, and start vespa on all nodes

**Comment by JayGhiya** [comment]
understood will be taking this up tomr and try to release some sort of draft. apologies for delayed response. been caught up

**Comment by JayGhiya** [comment]
hey @kkraune sorry for delay on this. engaged in multiple dev activities. will be taking a look post 1-2 weeks.

**Comment by JayGhiya** [comment]
This will still be delayed as caught up in multiple oss dev activities
